### DISCUSSION TITLE
Difference (duration) of two xsd:dateTime literals

### DISCUSSION THREAD

**Discussion by arrascue** [eclipse-rdf4j/rdf4j#4780] [Q&A] [answered]
Hello,

Is there a way to get the difference between two date times in RDF4J?

```
SELECT *
WHERE {
    bind("2023-09-14T06:37:46.225Z"^^xsd:dateTime as ?t1)
    bind("2025-07-29T10:06:59.109Z"^^xsd:dateTime as ?t2)
    bind(?t2 - ?t1 as ?diff)
}
```

?diff returns empty.

Thank you very much

**Comment by hmottestad** [comment, marked as answer]
I'm not sure if this is possible. The SPARQL standard only seems to support numbers when using subtraction. 

You can use < and > to check which is smaller, but it won't give you the diff. 

I couldn't find an epoch function either. Which would have made it possible to convert the date time to a number you could diff.

**Reply by hmottestad** [reply]
You could implement a custom function if you want: https://rdf4j.org/documentation/tutorials/custom-sparql-functions/

**Reply by arrascue** [reply]
Thank you!
